### PR TITLE
bump version of go to 1.19

### LINKED ISSUE #4966: Upgrade to Go 1.18 or higher
After the release of Go 1.19, Go 1.17 has fallen out of support. We need to upgrade to Go 1.18 or higher to receive the latest security fixes.

### PR DESCRIPTION
**Pull request by steeling**
Resolves #4966

### CODE REVIEW

**Comment by keithmattix** [comment]
@steeling Can you please link the associated issue?

File: pkg/trafficpolicy/trafficpolicy.go [hunk 1]
```diff
@@ -9,7 +9,6 @@ import (
 	hashstructure "github.com/mitchellh/hashstructure/v2"
 
 	policyv1alpha1 "github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
-
```
Comment (keithmattix) [inline pkg/trafficpolicy/trafficpolicy.go:12]: I think we still want this space here to separate locally generated k8s resources from app code

**Review by keithmattix — APPROVED** [review]
LGTM. One, non-blocking comment

File: pkg/trafficpolicy/trafficpolicy.go (hunk 1 above)
Comment (steeling) [reply pkg/trafficpolicy/trafficpolicy.go]: done

**Review by jaellio — APPROVED** [review]

**Review by shashankram — APPROVED** [review]

**Pull request merged**